MdePkg/Include/IndustryStandard/PeImage.h

```diff
@@ -233,7 +233,7 @@ typedef struct {
   UINT64                      SizeOfHeapCommit;
   UINT32                      LoaderFlags;
   UINT32                      NumberOfRvaAndSizes;
-  EFI_IMAGE_DATA_DIRECTORY  DataDirectory[EFI_IMAGE_NUMBER_OF_DIRECTORY_ENTRIES];
+  EFI_IMAGE_DATA_DIRECTORY    DataDirectory[EFI_IMAGE_NUMBER_OF_DIRECTORY_ENTRIES];

```

Comment:
It requires less change* to the debug tools if the new `EFI_IMAGE_NT_HEADERS32` and `EFI_IMAGE_NT_HEADERS64` in `PeImage2.h` have `EFI_IMAGE_DATA_DIRECTORY DataDirectory[EFI_IMAGE_NUMBER_OF_DIRECTORY_ENTRIES]` (as in the old header, here), instead of the new `EFI_IMAGE_DATA_DIRECTORY DataDirectory[]`. But I gather, from trying to change it, that there are 'reasons' (such as that it breaks the tools) for them to be `[]` now.

*Specifically, e.g. in `gdb_uefi.py` I used to be able to use `sections = (opt.address + 1).cast(sect_t)`, whereas now I need `sections = (opt['DataDirectory'][0].address + 16).cast(sect_t)`.

Reply:
Adding a related point, I don't believe there's any way to extract the '16' value from the rest of the info in the struct - so the previous version, which hardcodes the array length as 16 at least (implicitly) includes this info about the length of DataDirectory, whereas the current version doesn't include it at all ... I believe? :-/